src/GoTrueClient.ts

```diff
@@ -709,7 +731,7 @@ export default class GoTrueClient {
     }
 
     const hasExpired = currentSession.expires_at
-      ? currentSession.expires_at <= Date.now() / 1000
+      ? currentSession.expires_at - EXPIRY_MARGIN <= Date.now() / 1000
```

Comment:
hmm i rmb we didn't do this before because there were some other places where we used `getSession` and checked the expiry margin too which would result in double counting - but just need to check again to be sure

Reply:
You're right there was an issue... I'll actually add the `Date.now()` in the param so when called from `_recoverAndRefresh` it will use the current time + margin, and otherwise just the current time.